plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -117,6 +120,8 @@
     private final GlueMetastoreStats stats;
 
     private final Map<SchemaTableName, TableMetadata> tableMetadataCache = new ConcurrentHashMap<>();
+    private final Map<SchemaTableName, ConnectorViewDefinition> viewCache = new ConcurrentHashMap<>();
+    private final Map<SchemaTableName, ConnectorMaterializedViewDefinition> materializedViewCache = new ConcurrentHashMap<>();
```

Comment:
this & tableMetadataCache -- they should never be out of sync.
maybe we should have a single cache with `T | V | MV` as the value?

Reply:
I can try that, but what do you mean by "out of sync"?

Reply:
Technically, it's possible that two or three caches contain an entry with same key.
This would lead to inconsistent state (eg we would report given name to be a table, and also a view), since order of checking these caches is different in respective methods.

Reply:
> a single cache with T | V | MV as the value

i don't know whether this would be pretty or ugly.
maybe fixing the ordering to be the same would be enough

Reply:
I added some more verification, but I don't think this can happen. Anywhere the caches are set has the pattern:
- Check if the relevant cache has the key, if so return
- Check if any of the other caches has an entry, if so return
- Get and cache the entry

Reply:
i think it's fine